packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -565,6 +592,30 @@ describe('TransactionController', () => {
               blockTracker: finalNetwork.blockTracker,
               provider: finalNetwork.provider,
             };
+          case 'sepolia':
+            return {
+              configuration: {
+                chainId: SEPOLIA.chainId,
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
+            };
+          case 'goerli':
+            return {
+              configuration: {
+                chainId: GOERLI.chainId,
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
+            };
+          case 'customNetworkClientId-1':
+            return {
+              configuration: {
+                chainId: '0xa',
+              },
+              blockTracker: buildMockBlockTracker('0x1'),
+              provider: MAINNET_PROVIDER,
```

Comment:
do these provider and block tracker mocks need to be different instances, or would we rather deal with that later?

Reply:
i think we need to have different provider references and also add spys/assertions on exactly which provider instance was used in instantiation of the trackers. Expect one problem. You wouldn't know if the provider was assigned to the trackers for the right networkClientId. So even if we add a spy to the trackers constructors, you would only know that a particular provider was used and how many times it was used.

Let's just punt on this